### PR TITLE
bpo-38706: Replace "is true" with "is a truthy value" and "is false" with "is a falsey value".

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
https://bugs.python.org/issue38706

### CODE REVIEW

File: Doc/library/typing.rst [hunk 1]
```diff
@@ -365,7 +365,7 @@ need to mix dynamically and statically typed code.
 
 Contrast the behavior of :data:`Any` with the behavior of :class:`object`.
 Similar to :data:`Any`, every type is a subtype of :class:`object`. However,
-unlike :data:`Any`, the reverse is not true: :class:`object` is *not* a
+unlike :data:`Any`, the reverse is not a truthy value: :class:`object` is *not* a
```
Comment (ilevkivskyi) [inline Doc/library/typing.rst:368]: This is why `sed` may be a bad idea. This should stay "true".

**Review by ilevkivskyi — CHANGES_REQUESTED** [review]

File: Doc/library/typing.rst (hunk 1 above)
Comment (serhiy-storchaka) [reply Doc/library/typing.rst:368]: I have doubts about all other changes too. This PR is just a demonstration how large the change can be. And it is only a small part, there are around 1000 other potential changes, which must be made and checked manually.

File: Doc/library/dataclasses.rst [hunk 2]
```diff
@@ -172,7 +172,7 @@ Module-level decorators, classes, and functions
      def __init__(self, a: int, b: int = 0):
 
    :exc:`TypeError` will be raised if a field without a default value
-   follows a field with a default value.  This is true either when this
+   follows a field with a default value.  This is a truthy value either when this
```
Comment (ericvsmith) [inline Doc/library/dataclasses.rst:175]: This shouldn't change.

**Review by ericvsmith — COMMENTED** [review]
I marked one instances in dataclasses.rst that shouldn't change. I'm not sure the others are an improvement, either. I understand from another comment that this is just a demonstration.

**Comment by terryjreedy** [comment]
-1000  One of the two idlelib instances of 'is true' does not refer to a Python boolean value.

**Comment by stevendaprano** [comment]
Serhiy, nobody asked for every single copy of "is true" to be changed across the entire code base. Nobody wants that, nobody requested that, and it is a terrible, terrible idea to mechanically do a global search-and-replace over the entire code base.

I am sure that you know this is a terrible idea. I am trying very hard to assume good faith, but honestly I cannot see any possibility that you made this patch in good faith expecting it to be accepted. I think you made this patch in order to poison the discussion on b.p.o #38706.

The bug report wasn't about doing a global search and replace and blindly replacing every single copy of "is true". The bug report was to improve two specific error messages in two related methods so that they are more understandable less likely to cause misunderstandings.

**Comment by michael-carbon** [comment]
I would like to add some more points in this topic as I created the issue in the first place having the next three things in mind. 

a) The words True and False in those two functions are used in multiple contexts. Namely, 
**Name of functions** : in the function name `assertTrue`, does True refer to true and false in the context of english or True and False in the context of python? 
**Error message **: 'False is not true', I think it is safe to assume that both of `False` and `true` here in the same context between them, but I am not sure if this is the english language context or python's context. 

b) The error message does not inform me actually that expression (passed in as assertTrue's argument) is true, it is checking its truthiness and then lets me know that  'False is not true' which is a true statement *in English*, but it is not telling me that the expression i passed in is true or false. 

c) The suggested error message  'False is not True', is true in python *and* in english.

**Review by gpshead — CHANGES_REQUESTED** [review]
This PR must not go in.

In English "is true" and "is false" is correct.  "is a truthy value" and "is a falsey value" are not correct.

truthy and falsey are not words.  We should not use them in the documentation, everyone using python needs to understand that "is true" in documentation means "evaluates as true using python semantics of checking for a `__bool__`, etc, etc, etc per the Python truth testing protocol"

We should not state that everywhere in the text that the term true or false is used.

**Comment by ezio-melotti** [comment]
I thought the convention was to just use lowercase true/false to indicate truthiness and falseness, and `True`/`False` to indicate the constants.  This is what I've been following for years, even though I don't think it's explicitly documented anywhere.
 
@michael-carbon 
This came up in the past and is now documented clearly: https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertTrue
The docstring and the error message also use the lowercase true/false.

**Pull request closed without merging**